### PR TITLE
new: add retry logic for model downloading

### PR DESCRIPTION
**Pull request by joein**

### CODE REVIEW

File: fastembed/common/model_management.py [hunk 1]
```diff
@@ -233,26 +224,37 @@ def download_model(cls, model: Dict[str, Any], cache_dir: Path, **kwargs) -> Pat
...
+            if url_source:
+                try:
+                    return cls.retrieve_model_gcs(model["model"], url_source, str(cache_dir))
+                except Exception:
+                    logger.error(f"Could not download model from url: {url_source}")
+
+            logger.error(
+                f"Could not download model from either source, sleeping for {sleep} seconds, {retries} retries left."
+            )
+            sleep *= 3
```
Comment (Anush008) [inline fastembed/common/model_management.py:258]: `sleep` is unused?

Comment (joein) [reply fastembed/common/model_management.py:258]: lol, right, thank you!

Comment (joein) [reply fastembed/common/model_management.py]: fixed

**Review by Anush008 — APPROVED** [review]
Works! Thank you.

**Pull request merged**